concordium-std/src/impls.rs

```diff
@@ -578,12 +633,7 @@ impl Iterator for ContractStateIter {
     type Item = StateEntry;
 
     fn next(&mut self) -> Option<Self::Item> {
-        let res = unsafe { next(self.iterator_id) };
-        if res < 0 {
-            None
-        } else {
-            Some(StateEntry::open(res as u32))
-        }
+        self.state_iterator_next().unwrap_abort()
```

Comment:
That depends a bit on what the intent of this `Iterator` implementation is.

One option is to abort, but this seems like the worst option.
Another option is to just return None, terminate the iterator when encountering an error.

The last option is to have an iterator over `Result<StateEntry, _>`. This would then reflect the fact that this is just a convenience wrapper around `state_iterator_next`.

I would imagine this is only used in low-level implementations, so my preference would be option (3), or remove the implementation of the Iterator trait entirely.